src/utils/event.ts

```diff
@@ -61,3 +62,26 @@ export function warpEventCanonicalSignaturehash(eventName: string, argTypes: arg
     BigInt(`0x${createKeccakHash('keccak256').update(funcSignatureHash).digest('hex')}`) & MASK_250
   ).toString(16)}`;
 }
+
+// NOTE: argTypes must not contain `uint` , it should be `uint256` instead
+export function warpEventCanonicalSignaturehash256(
+  eventName: string,
+  argTypes: argType[],
+): { low: string; high: string } {
+  const getArgStringRepresentation = (arg: argType): string => {
+    if (typeof arg === 'string') return arg;
+    return `(${arg.map(getArgStringRepresentation).join(',')})`;
+  };
+
+  const funcSignature = `${eventName}(${argTypes.map(getArgStringRepresentation).join(',')})`;
+  const funcSignatureHash = createKeccakHash('keccak256').update(funcSignature).digest('hex');
+
+  const hash = BigInt(`0x${createKeccakHash('keccak256').update(funcSignatureHash).digest('hex')}`);
```

Comment:
what is this hash, we want to be as close to ethereum's encoding.
just the hash of the function signature is enough

Reply:
It went unnoticed in the upper function also